src/Compilers/CSharp/Portable/Syntax/SyntaxBindingUtilities.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax
+{
+    internal static class SyntaxBindingUtilities
+    {
+        public static bool BindsToResumableStateMachineState(SyntaxNode node)
+            => node.IsKind(SyntaxKind.YieldReturnStatement) ||
+               node.IsKind(SyntaxKind.AwaitExpression) ||
+               node is CommonForEachStatementSyntax { AwaitKeyword.RawKind: not 0 }
+                    or VariableDeclaratorSyntax { Parent.Parent: UsingStatementSyntax { AwaitKeyword.RawKind: not 0 } or LocalDeclarationStatementSyntax { AwaitKeyword.RawKind: not 0 } }
+                    or UsingStatementSyntax { Expression: not null, AwaitKeyword.RawKind: not 0 };
+
+        public static bool BindsToTryStatement(SyntaxNode node)
```

Comment:
>BindsToTryStatement [](http://example.com/codeflow?start=27&length=19)

Is this equivalent to the set of `IsKind()` checks previously at the callsites? If so, is there a reason to use this form? (The previous form was simpler to read.)

Reply:
It's not equivalent. We need to check not only the node kinds but also structure of the nodes for some nodes.